include/deal.II/numerics/solution_serialization.h

```diff
@@ -0,0 +1,226 @@
+  }
+
+  void
+  add_vectors(std::vector<VectorType *> vectors)
+  {
+    for (auto v : vectors)
+      this->vectors.emplace_back(v);
+  }
+
+  void
+  add_vector(const VectorType &vector)
+  {
+    this->vectors.emplace_back(const_cast<VectorType *>(&vector));
+  }
+
+  void
+  add_vectors(std::vector<const VectorType *> vectors)
+  {
+    for (auto v : vectors)
+      this->vectors.emplace_back(const_cast<VectorType *>(v));
+  }
+
+  void
+  save(const std::string file_name)
+  {
+    // determine local size
+    const size_type local_size = n_locally_owned_dofs * vectors.size();
+
+    // collect all values in a single vector
+    std::vector<value_type> temp(local_size);
+
+    for (unsigned int b = 0, c = 0; b < vectors.size(); ++b)
+      for (unsigned int i = 0; i < n_locally_owned_dofs; ++i, ++c)
+        temp[c] = vectors[b]->local_element(unique_dof_map[i]);
+
+    // write to hard drive
+    mpi_io_read_or_write(false /*write*/, file_name, temp);
+  }
+
+  void
+  load(const std::string file_name)
+  {
+    // determine local size
+    const size_type local_size = n_locally_owned_dofs * vectors.size();
+
+    // read from hard drive
+    std::vector<value_type> temp(local_size);
+    mpi_io_read_or_write(true /*read*/, file_name, temp);
+
+    // split up single vector into blocks
+    for (unsigned int b = 0, c = 0; b < vectors.size(); ++b)
+      for (unsigned int i = 0; i < n_locally_owned_dofs; ++i, ++c)
+        vectors[b]->local_element(unique_dof_map[i]) = temp[c];
+  }
+
+private:
+  const MPI_Comm                  comm;
+  const size_type                 n_locally_owned_dofs;
+  const std::vector<unsigned int> unique_dof_map;
+  std::vector<VectorType *>       vectors;
+
+  static void
+  visit_cells_recursevely(
+    const typename DoFHandler<dim, spacedim>::cell_iterator &cell,
+    const IndexSet &                                         locally_owned_dofs,
+    std::vector<bool> &                                      mask,
+    std::vector<unsigned int> &                              result)
+  {
+    if (cell->is_active())
+      {
+        if (cell->is_locally_owned())
+          {
+            std::vector<types::global_dof_index> dof_indices(
+              cell->get_fe().n_dofs_per_cell());
+
+            cell->get_dof_indices(dof_indices);
+
+            for (const auto &i : dof_indices)
+              {
+                if (locally_owned_dofs.is_element(i) == false)
+                  continue;
+
+                const auto i_local = locally_owned_dofs.index_within_set(i);
+
+                if (mask[i_local] == true)
+                  continue;
+
+                mask[i_local] = true;
+                result.push_back(i_local);
+              }
+          }
+      }
+    else
+      {
+        for (const auto &child : cell->child_iterators())
+          visit_cells_recursevely(child, locally_owned_dofs, mask, result);
+      }
+  }
+
+  std::vector<unsigned int> static create_unique_dof_map(
```

Comment:
The challenge was that the indices of active cells different depending on the refinement history even though the mesh might be identical. The consequence is that the DoFs also get different indices. This is fixed by introducing an intermediate enumeration, which depends on the id of the coarse cell.